src/rocksdb.rs

```diff
@@ -417,6 +436,67 @@ impl DB {
         })
     }
 
+    pub fn open_for_read_only(
+        opts: DBOptions,
+        path: &str,
+        error_if_log_file_exist: bool,
+    ) -> Result<DB, String> {
+        Self::open_cf_for_read_only(opts, path, vec![], error_if_log_file_exist)
+    }
+
+    pub fn open_cf_for_read_only(
```

Comment:
seem we can use most of the same codes for open and open read-only?

Reply:
Thanks for your reply. 
You mean "need more cleanup" ?
I'll try it.

Reply:
RFC:

I think `open_cf` arguments should be

```rust
pub fn open_cf(
    opts: DBOptions,
    path: &str,
    cfs: Vec<(&str, ColumnFamilyOptions)>,
) -> Result<DB, String> {
// ...
}
```

not

```rust
pub fn open_cf(
    opts: DBOptions,
    path: &str,
    cfs: Vec<&str>,
    cf_opts: Vec<ColumnFamilyOptions>,
) -> Result<DB, String> {
// ...
}
```

Later is error prone, so I want to fix this.
Do you have any objections to this change?

Reply:
I prefer not using tuple but a ColumnFamilyDescriptor like RocksDB. But this change may break the compatibility but can make the open function more elegant. 

/cc @BusyJay @zhangjinpeng1987

Reply:
Like this?

```rust
pub fn open_cf<T: Into<ColumnFamilyDescriptor>>(
    opts: DBOptions,
    path: &str,
    cfs: Vec<T>,
) -> Result<DB, String> {
// ...
}
```

Reply:
I prefer `&[T]`  :-)

Reply:
Shall we just add a `read_only` parameter for `open_cf`?

Reply:
I've add a method `open_cf_internal`,
every other `open` method invoke this.

```rust
fn open_cf_internal<'a, T>(
    opts: DBOptions,
    path: &str,
    cfds: Vec<T>,
    // if none, open for read write mode.
    // otherwise, open for read only.
    error_if_log_file_exist: Option<bool>,
) -> Result<DB, String>
where
    T: Into<ColumnFamilyDescriptor<'a>>,
{
    // ...
}
```

I'm trying to change `Vec<T>` to `&[T]` now.

Reply:
@siddontang @zhangjinpeng1987 
Changing `Vec<T>` to `&[T]` may need to change type of `DB`
because of `ColumnFamilyOptions`'s implementation of `Clone` and  lifetime.

I think `Vec<T>` is better.

Reply:
Got it.